make/CheckJavaOSVersion.java

```diff
@@ -34,7 +34,7 @@ public static void main(String... args) {
 
     private static void checkJavaOSVersion(String expectVersion) {
         String osVersion = System.getProperty("os.version");
-        if (!osVersion.equals(expectVersion)) {
+        if (!osVersion.startsWith(expectVersion)) {
```

Comment:
This change seems questionable.

`os.version` is supposed to reflect (`.equals`) the OS version. We should at least understand why it is being reporting as a leading substring.   
Instead of `.startsWith`, consider `.matches` with a pattern that allows optional trailing dotted digits.

Reply:
Could be related to https://bugs.openjdk.java.net/browse/JDK-8269850

The [error message](https://github.com/openjdk/jtreg/runs/6034597187?check_suite_focus=true#step:6:197) reported in the logs reads:
```text
/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.14-101/x64/Contents/Home/bin/java CheckJavaOSVersion.java 11.6
The version of JDK you are using does not report the OS version correctly.
    java.home:    /Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.14-101/x64/Contents/Home
    java.version: 11.0.14.1
    os.version:   11.6.5  (expected: 11.6)
Use a more recent update of this version of JDK, or a newer version of JDK.
make: *** [checkJavaOSVersion] Error 1
```

Thus, `11.6.5` did not match `11.6`.

Reply:
Added an explanatory comment.